integration-tests/src/main/java/io/quarkus/it/amazon/dynamodb/DynamoDBEnhancedResource.java

```diff
@@ -0,0 +1,87 @@
+package io.quarkus.it.amazon.dynamodb;
+
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.jboss.logging.Logger;
+
+import software.amazon.awssdk.enhanced.dynamodb.*;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+@Path("/dynamodbenhanced")
+public class DynamoDBEnhancedResource {
+    private final static String ASYNC_TABLE = "async";
+    private final static String BLOCKING_TABLE = "blocking";
+
+    private static final Logger LOG = Logger.getLogger(DynamoDBEnhancedResource.class);
+
+    @Inject
+    DynamoDbClient dynamoClient;
+
+    @Inject
+    DynamoDbAsyncClient dynamoAsyncClient;
+
+    @GET
+    @Path("async")
+    @Produces(TEXT_PLAIN)
+    public CompletionStage<String> testAsyncDynamo() throws InterruptedException {
+        LOG.info("Testing Async Dynamodb client with table: " + ASYNC_TABLE);
+        String partitionKeyAsString = UUID.randomUUID().toString();
+
+        DynamoDbEnhancedAsyncClient enhancedAsyncClient = DynamoDbEnhancedAsyncClient.builder()
```

Comment:
I guess, the instance should be provided by the extension

Reply:
What do you mean? According to the aws v2 sdk a DynamoDBEnhancedClient can use an underlying DynamoDBClient. Isn't it safer to follow the sdk way? Could you elaborate what you mean?

Reply:
I mean a producer so you can just inject the enhanced version like the other one. I guess the user only need one version at a time or is it design so that you need both version to cover all sdk api ?

Reply:
I think you need both if you want to cover all sdk (low-level, high-level). I also do not know if the enhanced client replaces the other client. To be honest, I don't think so. They are even in seperate locations (https://github.com/aws/aws-sdk-java-v2/tree/master/services/dynamodb vs https://github.com/aws/aws-sdk-java-v2/tree/master/services-custom/dynamodb-enhanced).